src/System.Console/src/System/ConsolePal.Windows.cs

```diff
@@ -85,28 +86,52 @@ private static unsafe bool ConsoleHandleIsWritable(IntPtr outErrHandle)
             return r != 0; // In Win32 apps w/ no console, bResult should be 0 for failure.
         }
 
-        // Note if we ever support different encodings:
-        // We always use file APIs in WindowsConsoleStream since WriteConsole is called only when the Encoding is Unicode and 
-        // the handle is not redirected. Since changing the Input/OutputEncoding is not currently supported, we will always have 
-        // the either the GetConsoleCP encoding or the UTF8Encoding fallback, in which case we always use the Read/WriteFile native
-        // API.  If that ever changes, WindowsConsoleStream will need to be changed, too.
-
         public static Encoding InputEncoding
         {
-            get { return GetEncoding((int)Interop.mincore.GetConsoleCP()); }
+            get { return EncodingHelper.GetSupportedConsoleEncoding((int)Interop.mincore.GetConsoleCP()); }
+        }
+
+        public static void SetConsoleInputEncoding(Encoding enc)
+        {
+            if (enc.CodePage != Encoding.Unicode.CodePage)
```

Comment:
Again it in unclear. The native APIs support code page setting to Unicode. My guess is it is done because not all fonts support wide chars display, however that still does not explain why only Unicode is ignored. I let this be for app-compat reasons.